deno/generate_short_url/src/mod.ts

```diff
@@ -0,0 +1,67 @@
+export default async function (req: any, res: any) {
+  const { provider, url } = JSON.parse(req.payload);
+
+  if (!provider) {
+    res.json({ success: false, message: "provider is required" });
+    return;
+  }
+
+  if (!url) {
+    res.json({ success: false, message: "url is required" });
+    return;
+  }
+
+  if (provider === "bitly") {
+    const BITLY_TOKEN = req.variables["BITLY_TOKEN"];
+
+    if (!BITLY_TOKEN) {
+      res.json({ success: false, message: "BITLY_TOKEN is required" });
+      return;
+    }
+
+    const response = await fetch(`https://api-ssl.bitly.com/v4/shorten`, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${BITLY_TOKEN}`,
+      },
+      body: JSON.stringify({
+        long_url: url,
+      }),
+    });
+
+    if (response.status !== 200) {
+      res.json({ success: false, message: `${await response.text()}` });
+      return;
+    }
```

Comment:
I'm getting a failure for successful shortening (probably because the response status is 201.

```json
{
  "response": {
    "success": false,
    "message": "{\"created_at\":\"2022-11-22T23:28:44+0000\",\"id\":\"bit.ly/3XnGL5n\",\"link\":\"https://bit.ly/3XnGL5n\",\"custom_bitlinks\":[],\"long_url\":\"https://google.com/\",\"archived\":false,\"tags\":[],\"deeplinks\":[],\"references\":{\"group\":\"https://api-ssl.bitly.com/v4/groups/Ba89eOlNFki\"}}"
  },
  "stdout": "",
  "stderr": ""
}
```

Reply:
Have made the required fix.